packages/react-vapor/src/components/validation/hoc/WithNonEmptyValueMultiSelectValidationHOC.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {createStructuredSelector} from 'reselect';
+import {IDispatch} from '../../../utils/ReduxUtils';
+import {IMultiSelectOwnProps} from '../../select/MultiSelectConnected';
+import {SelectSelector} from '../../select/SelectSelector';
+import {ValidationActions} from '../ValidationActions';
+import {ValidationTypes} from '../ValidationTypes';
+import {InferableComponentEnhancer} from './connectHOC';
+
+const mapStateToProps = createStructuredSelector({
+    selectedValues: SelectSelector.getMultiSelectSelectedValues,
+});
+
+const mapDispatchToProps = (dispatch: IDispatch) => ({
+    setError: (id: string, error: string) => dispatch(ValidationActions.setError(id, error, ValidationTypes.nonEmpty)),
+    clearError: (id: string) => dispatch(ValidationActions.clearError(id, ValidationTypes.nonEmpty)),
+});
+
+export interface IWithNonEmptyValueMultiSelectValidationProps {
+    validationMessage?: string;
+    resetErrorOnUnmount?: boolean;
+    isException?: boolean;
```

Comment:
What is the use case of `isException`? If set to true, the component won't do anything 🤔 Shouldn't it always be `false`, thus not really useful?

Reply:
I added this since we use the same form to create and update a catalog. This means when someone updates a catalog that has no sources, they would not be able to do so. In the admin-ui part the only case where they would not have the error would be if the catalog exists && there were no initial source selected

Reply:
Still not keen on adding a boolean to disable the whole feature of the HOC. I feel like we should just not include the HOC if we don't need its validation.

I think it might be done by creating a simple wrapping component that either use the HOC or not. Something like the following (not accurate, but should explain the idea :P ) 

```
const MultiSelectWithValidation = nonEmpty(withDirty(MultiSelect));
const MultiSelectWithout = withDirty(MutliSelect);
const SourceIdsMultiSelect = ({ isEditing, ...props }) => isEditing ? <MultiSelectWithout {...props} /> : <MultiSelectWithValidation {...props} />;
```